primitives/src/locktime/relative.rs

```diff
@@ -25,8 +23,7 @@ pub use units::locktime::relative::{Height, Time, TimeOverflowError};
 /// ### Note on ordering
 ///
 /// Locktimes may be height- or time-based, and these metrics are incommensurate; there is no total
-/// ordering on locktimes. We therefore have implemented [`PartialOrd`] but not [`Ord`]. We also
-/// implement [`ordered::ArbitraryOrd`] if the "ordered" feature is enabled.
+/// ordering on locktimes. We therefore have implemented [`PartialOrd`] but not [`Ord`].
```

Comment:
I hope we did not. `PartialOrd` still allows `<` & related operators.

Reply:
Agreed. Unfortunately `PartialOrd`, while being conceptually exactly what we want, still has language hooks into the comparison operators that make it more-or-less useless for this purpose.

Reply:
Damn, I misunderstood how Rust works. Now I get it. In case this snippet is useful to anyone
```rust
        let lock_by_height = LockTime::from_height(144); // 144 blocks, approx 24h.
        let lock_by_time = LockTime::from_512_second_intervals(168); // 168 time intervals, approx 24h.

        // Sanity checks.
        assert!(lock_by_height.is_block_height());
        assert!(lock_by_time.is_block_time());

        // PartialOrd allows users to write this and its plain wrong.
        if lock_by_height < lock_by_time {
            unreachable!()
        } {
            // We got `None` but users might interpret this as lock_by_height does not satisfy.
        }
```

Reply:
I mean, strictly speaking it's not "plain wrong". In this case both `lock_by_height < lock_by_time` and `lock_by_time < lock_by_height` will be `false` (as will every other combination, using `>` or `<=` or `==` or whatever).

But people expect that `<` is exactly the opposite of `>=`, which is why this is dangerous. I've never seen code that checks if `a < b`, and then failing that, confirm that indeed `a >= b`.

This sort of "technically mathematically correct, but will be used wrong 100% of the time in practice" thing maybe appears to C compiler authors ... and it's not even consistent with the rest of Rust's stdlib which is mathematically a mess.